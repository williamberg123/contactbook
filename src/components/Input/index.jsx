import PropTypes from 'prop-types';
import StyledInput from './styles';

export default function Input({ type, placeholder, elementRef }) {
	return (
		<StyledInput required ref={elementRef} type={type} placeholder={placeholder} />
	);
}

Input.propTypes = {
	type: PropTypes.string.isRequired,
	placeholder: PropTypes.string,
	elementRef: PropTypes.instanceOf(Object),
};
