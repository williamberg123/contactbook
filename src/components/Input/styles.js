import styled from 'styled-components';

const StyledInput = styled.input`
	padding: 15px;
`;

export default StyledInput;
