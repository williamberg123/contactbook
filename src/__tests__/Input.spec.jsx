import { useRef } from 'react';
import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import PropTypes from 'prop-types';
import Input from '../components/Input';

const Container = ({ inputInfo }) => {
	const elementRef = useRef();

	return (
		<div>
			<Input {...inputInfo} elementRef={elementRef} />
		</div>
	);
};

Container.propTypes = {
	inputInfo: PropTypes.shape({
		type: PropTypes.string.isRequired,
		placeholder: PropTypes.string.isRequired,
	}).isRequired,
};

describe('<Input />', () => {
	const inputInfo = {
		type: 'number',
		placeholder: 'my input element',
	};

	it('should render Input correctly', () => {
		render(<Container inputInfo={inputInfo} />);

		expect(screen.getByPlaceholderText(inputInfo.placeholder)).toBeInTheDocument();
		expect(screen.getByPlaceholderText(inputInfo.placeholder)).toHaveAttribute('type', 'number');
	});

	it('should match snapshot', () => {
		render(<Container inputInfo={inputInfo} />);
		expect(screen.getByPlaceholderText(inputInfo.placeholder)).toMatchSnapshot();
	});
});
