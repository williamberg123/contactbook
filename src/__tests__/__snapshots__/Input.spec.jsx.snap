// Vitest Snapshot v1

exports[`<Input /> > should match snapshot 1`] = `
.c0 {
  padding: 15px;
}

@media (max-width:600px) {
  .c0 {
    padding: 10px;
  }
}

<input
  class="c0"
  placeholder="my input element"
  required=""
  type="number"
/>
`;
